src/dbnode/storage/index/block.go

```diff
@@ -797,12 +808,12 @@ func (b *block) pooledID(id []byte) ident.ID {
 func (b *block) addAggregateResults(
 	cancellable *xresource.CancellableLifetime,
 	results AggregateResults,
-	batch []AggregateResultsEntry,
+	batch AggregateResultsEntries,
 	source []byte,
-) ([]AggregateResultsEntry, int, int, error) {
+) (AggregateResultsEntries, int, int, error) {
 	// update recently queried docs to monitor memory.
 	if results.EnforceLimits() {
-		if err := b.docsLimit.Inc(len(batch), source); err != nil {
+		if err := b.docsLimit.Inc(batch.Size(), source); err != nil {
```

Comment:
Will have to make a decision on how docs limit is affected in aggregate results; this is not correct